design/capm3-remediation-controller-proposal.md

```diff
@@ -104,24 +104,28 @@ Metal3RemediationTemplate.
 #### Remediation Controller
 
 * MRC watches for the presence of Metal3Remediation CR.
-* Based on the remediation strategy defined in ```.spec.strategy``` in
+* Based on the remediation strategy defined in ```.spec.strategy.type``` in
   Metal3Remediation, RC uses BMO APIs to get hosts back into a healthy or
   manageable state.
 * RC uses ```.status.phase``` to save the states of the remediation. Available
-  states are running, rebooting, power off, failed.
+  states are running, waiting, deleting machine.
 * After RC have finished its remediation, it will wait for the Metal3Remediation
   CR to be removed. (When using CAPI MachineHealthCheck controller,
   MHC will noticed the Node becomes healthy and deletes the instantiated
   MachineRemediation CR.).
-* If RCs timeout for Node to become healthy expires, it annotates BareMetalHost
-  with ```capi.metal3.io/unhealthy``` annotation, sets BMH Online field
-  to False and removes the Machine object.
+* If RCs timeout for Node to become healthy expires, it sets
+  `capi.MachineOwnerRemediatedCondition` to False on Machine object
```

Comment:
Whats the motivation here if we immediately remove the Machine anyway?

Reply:
We need to use `capi.MachineOwnerRemediatedCondition` to be able to owning machine controller to do the deletion of the Machine object. For example, if our Machine is control plane CAPI will handle preflight checks, keep etcd quorum healthy and deletes Machine after needed healtchecks. If we just set the deletion timestamp on Machine we are able to break the controlplane and we have high risk to lose the whole cluster.

Reply:
So we are not removing the Machine ourselves we let CAPI to handle the removing of the Machines.

Reply:
Fair enough. Although realistically, if we're deleting the machine then chances are it is already /not/ contributing towards the etcd quorum calculation.